sway-lib-core/src/ops.sw

```diff
@@ -618,15 +618,15 @@ impl Shiftable for b256 {
 /////////////////////////////////////////////////
 
 /// Left shift a u64 and preserve the overflow amount if any
-fn lsh_with_carry(word: u64, shift_amount: u64) -> (u64, u64) {
+pub fn lsh_with_carry(word: u64, shift_amount: u64) -> (u64, u64) {
```

Comment:
`core` is meant to just contain ops, necessary so `std` can use operators. Everything else in `core` should be private. If you need access to these two helper functions, then copy them into `std`.

The only other thing so far that's acceptable is `core::num`, to imitate Rust's https://doc.rust-lang.org/src/core/num/mod.rs.html#849-853 (and even then we should consider moving that to `std` as per https://doc.rust-lang.org/std/primitive.u64.html#associatedconstant.MAX).

There was a suggestion to add a second library in between `core` and `std`, but I can't find it. cc @nfurfaro